src/org/jetbrains/plugins/scala/codeInspection/functionExpressions/UnnecessaryPartialFunctionQuickFix.scala

```diff
@@ -0,0 +1,54 @@
+package org.jetbrains.plugins.scala.codeInspection.functionExpressions
+
+import com.intellij.openapi.project.Project
+import org.jetbrains.plugins.scala.codeInspection.functionExpressions.UnnecessaryPartialFunctionQuickFix._
+import org.jetbrains.plugins.scala.codeInspection.{AbstractFixOnPsiElement, InspectionBundle}
+import org.jetbrains.plugins.scala.lang.psi.ScalaPsiUtil
+import org.jetbrains.plugins.scala.lang.psi.api.base.patterns.{ScCaseClause, ScReferencePattern, ScWildcardPattern}
+import org.jetbrains.plugins.scala.lang.psi.api.expr.{ScBlock, ScBlockExpr}
+import org.jetbrains.plugins.scala.lang.psi.impl.ScalaPsiElementFactory
+
+object UnnecessaryPartialFunctionQuickFix {
+  val hint = InspectionBundle.message("convert.to.anonymous.function")
+}
+
+class UnnecessaryPartialFunctionQuickFix(caseClause: ScCaseClause, expression: ScBlockExpr)
+  extends AbstractFixOnPsiElement(hint, caseClause) {
+
+  override def doApplyFix(project: Project): Unit = {
```

Comment:
When you extend `AbstractFixOnPsiElement` you shouldn't access constructor parameters directly, they may be invalidated by the time the quickfix is invoked. It probably should be better documented, I'll fix it. You can get rid of `caseClause` parameter and use `getElement` to get `expression`. Also you need to check again that expression.caseClauses has a single case.